modules/navier_stokes/src/fvkernels/INSFVMixingLengthScalarDiffusion.C

```diff
@@ -0,0 +1,106 @@
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "INSFVMixingLengthScalarDiffusion.h"
+
+registerMooseObject("NavierStokesApp", INSFVMixingLengthScalarDiffusion);
+
+InputParameters
+INSFVMixingLengthScalarDiffusion::validParams()
+{
+  InputParameters params = FVFluxKernel::validParams();
+  params.addClassDescription("Computes the turbulent diffusive flux that appears in "
+                             "Reynolds-averaged fluid conservation equations.");
+  params.addRequiredCoupledVar("u", "The velocity in the x direction.");
+  params.addCoupledVar("v", "The velocity in the y direction.");
+  params.addCoupledVar("w", "The velocity in the z direction.");
+  params.addRequiredCoupledVar("mixing_length", "The turbulent mixing length.");
+  params.addRequiredParam<Real>(
+      "schmidt_number",
+      "The turbulent Schmidt number (or turbulent Prandtl number if the passive scalar is energy) "
+      "that relates the turbulent scalar diffusivity to the turbulent momentum diffusivity.");
+  params.set<unsigned short>("ghost_layers") = 2;
+  return params;
+}
+
+INSFVMixingLengthScalarDiffusion::INSFVMixingLengthScalarDiffusion(const InputParameters & params)
+  : FVFluxKernel(params),
+    _dim(_subproblem.mesh().dimension()),
+    _u_var(dynamic_cast<const INSFVVelocityVariable *>(
+        &_subproblem.getVariable(_tid, params.get<std::vector<VariableName>>("u").front()))),
+    _v_var(params.isParamValid("v")
+               ? dynamic_cast<const INSFVVelocityVariable *>(&_subproblem.getVariable(
+                     _tid, params.get<std::vector<VariableName>>("v").front()))
+               : nullptr),
+    _w_var(params.isParamValid("w")
+               ? dynamic_cast<const INSFVVelocityVariable *>(&_subproblem.getVariable(
+                     _tid, params.get<std::vector<VariableName>>("w").front()))
+               : nullptr),
+    _mixing_len(coupledValue("mixing_length")),
+    _schmidt_number(getParam<Real>("schmidt_number"))
+{
+#ifndef MOOSE_GLOBAL_AD_INDEXING
+  mooseError("INSFV is not supported by local AD indexing. In order to use INSFV, please run the "
+             "configure script in the root MOOSE directory with the configure option "
+             "'--with-ad-indexing-type=global'");
+#endif
+
+  if (!_u_var)
+    paramError("u", "the u velocity must be an INSFVVelocityVariable.");
+
+  if (_dim >= 2 && !_v_var)
+    paramError("v",
+               "In two or more dimensions, the v velocity must be supplied and it must be an "
+               "INSFVVelocityVariable.");
+
+  if (_dim >= 3 && !_w_var)
+    paramError("w",
+               "In three-dimensions, the w velocity must be supplied and it must be an "
+               "INSFVVelocityVariable.");
+}
+
+ADReal
+INSFVMixingLengthScalarDiffusion::computeQpResidual()
+{
+#ifdef MOOSE_GLOBAL_AD_INDEXING
+  constexpr Real offset = 1e-15; // prevents explosion of sqrt(x) derivative to infinity
+
+  // Compute the normalized velocity gradient.
```

Comment:
this is computing the norm

Can you please send me a reference for this derivation?

Reply:
ok saw the reference on the insfv page. Got to read that text book.

Reply:
This part is a bit of improv. All the textbook models I've seen are for problems with a clearly defined wall direction so they just take a gradient normal to the wall. I'm imagining that in the very near wall region, the gradient is very large in the perpendicular direction so the norm will approximate the perpendicular value. Not sure what to think about this model in the area far from the wall without doing some comparison to a benchmark.

Reply:
Well it's consistent with the other kernel and we'll have to tune it anyway so I think this is ok